docs/design/design_v1.9.md

```diff
@@ -69,196 +67,246 @@ This means we aim to standardize:
 
-### Generate Static Pod Manifest for local etcd
+### (optional) Write init kubelet configuration
 
-If the user specified an external etcd this step will be skipped, otherwise a static manifest file will be generated for creating a local etcd instance running in a pod with following attributes:
+If kubeadm is invoked with `--feature-gates=DynamicKubeletConfig`, it writes the kubelet init configuration into `/var/lib/kubelet/config/init/kubelet` file.
 
-- listen on `localhost:2379` and use `HostNetwork=true`
-- make a `hostPath` mount out from the `dataDir` to the host's filesystem
-- Any extra flags specified by the user
+The init configuration is used for starting the kubelet on this specific node, providing an alternative for the kubelet drop-in file; such configuration will be replaced by the kubelet base configuration as described in following steps.
 
 Please note that:
 
-1. The image, for version  `3.0.17` /current architecture, will be pulled from `gcr.io/google_containers`. In case an alternative image repository is specified this one will be used; In case an alternative image name is specified, this one will be used.
-2. in case of `kubeadm`  executed in the `--dry-run` mode, the etcd static pod manifest is written in a temporary folder.
-3. Static Pod Manifest generation for local etcd can be invoked individually with the `kubeadm phase etcd local` command.
+1. to make dynamic kubelet configuration work, flag `--dynamic-config-dir` should be specified in `/etc/systemd/system/kubelet.service.d/10-kubeadm.conf`.
 
 ### Wait for the control plane to come up
 
 This is a critical moment in time for kubeadm clusters.
 kubeadm waits until `localhost:6443/healthz` returns `ok`, however in order to detect deadlock conditions, kubeadm fails fast if `localhost:10255/healthz` (kubelet liveness) or `localhost:10255/healthz/syncloop` (kubelet readiness) don't return `ok`, respectively after 40 and 60 second.
 
-kubeadm relies on the kubelet to pull the control plane images and run them properly as Static Pods.
+kubeadm relies on the kubelet to pull the control plane images and run them properly as static Pods.
 But there are (as we've seen) a lot of things that can go wrong. Most of them are network/resolv.conf/proxy related.
 
 After the control plane is up, kubeadm completes a couple of tasks described in following paragraphs.
 
-### Saves MasterConfiguration in a ConfigMap for later reference
+### (optional) Write base kubelet configuration
+
+If kubeadm is invoked with `--feature-gates=DynamicKubeletConfig`: 
+
+1. Write the kubelet base configuration into the `kubelet-base-config-v1.9`  ConfigMap in the `kube-system` namespace.
+2. Creates RBAC rules for granting read access to such ConfigMap to all bootstrap tokens and all kubelets (`system:bootstrappers:kubeadm:default-node-token` and `system:nodes` groups).
+3. Update current node configuration specifying that the the source for the node/kubelet configuration is the above ConfigMap.
```

Comment:
Enable the dynamic kubelet configuration feature for the initial master node by pointing `Node.spec.configSource` to the newly-created configmap